core/src/main/java/org/springframework/security/authorization/method/MethodAccessDeniedHandler.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authorization.method;
+
+import org.springframework.lang.Nullable;
+import org.springframework.security.authorization.AuthorizationDecision;
+
+public interface MethodAccessDeniedHandler {
```

Comment:
Have you already considered making this a generic class, like `AuthorizationManager`? I think that I'd personally prefer the type safety of:

```java
Object handle(MethodInvocation mi, AuthorizationDecision decision) { 
    ....
}
```

in my implementation.